packages/datadog-plugin-jest/src/index.js

```diff
@@ -58,6 +64,88 @@ class JestPlugin extends Plugin {
     this.testEnvironmentMetadata = getTestEnvironmentMetadata('jest', this.config)
     this.codeOwnersEntries = getCodeOwnersFileEntries()
 
+    this.addSub('ci:jest:session:start', (command) => {
+      if (!this.config.isAgentlessEnabled) {
+        return
+      }
+      const store = storage.getStore()
+      const childOf = getTestParentSpan(this.tracer)
+      const testSessionSpanMetadata = getTestSessionCommonTags(command, this.tracer._version)
+
+      const testSessionSpan = this.tracer.startSpan('jest.test_session', {
+        childOf,
+        tags: {
+          ...this.testEnvironmentMetadata,
+          ...testSessionSpanMetadata
+        }
+      })
+      this.enter(testSessionSpan, store)
+    })
+
+    this.addSub('ci:jest:session:finish', (status) => {
+      if (!this.config.isAgentlessEnabled) {
+        return
+      }
+      const testSessionSpan = storage.getStore().span
+      testSessionSpan.setTag(TEST_STATUS, status)
+      testSessionSpan.finish()
+      finishAllTraceSpans(testSessionSpan)
+      this.tracer._exporter._writer.flush()
+    })
+
+    // Test suites can be run in a different process from jest's main one.
+    // This subscriber changes the configuration objects from jest to inject the trace id
+    // of the test session to the processes that run the test suites.
+    this.addSub('ci:jest:session:configuration', configs => {
+      if (!this.config.isAgentlessEnabled) {
+        return
+      }
+      const testSessionSpan = storage.getStore().span
+      configs.forEach(config => {
+        config._ddTestSessionId = testSessionSpan.context()._traceId.toString(16)
```

Comment:
Could these be stored outside of a variable from Jest like in a WeakMap?

Reply:
sadly no. This is how we inject this id to the worker processes that run the tests. Storing in a weak map here would do nothing, because it would be empty in the worker process